src/core/control/settings/ViewModes.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Xournal++
+ *
+ * View Mode struct & attribute definitions
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <string>           // for string
+#include <vector>           // for vector
+#include <unordered_map>    // for unordered_map
+
+// reserved default view mode ids
+constexpr auto VIEW_MODE_DEFAULT = 0;
+constexpr auto VIEW_MODE_FULLSCREEN = 1;
+constexpr auto VIEW_MODE_PRESENTATION = 2;
```

Comment:
I tried to avoid a dedicated enum, because of the future possibility to add custom view-modes.
But I guess when implementing this in a follow up one can change that and having an enum in this PR is probably the prettier solution.